### ISSUE TITLE
Ability to specify link media attribute

### ISSUE THREAD

**Issue by espositofulvio** [sveltejs/kit#3097]
### Describe the problem

For accessibility and performance it would be good to have a way to specify the media attribute for the generated CSS. 
I mean that all the 

<link rel="stylesheet" href="_app/index.012930190.css">

could be actually something like 

<link rel="stylesheet" href="_app/index.012930190.css" media="screen">

thus many screen or braille readers, etc., could avoid downloading the stylesheets.

### Describe the proposed solution

I'm not sure what a good a good solution would be but I guess I would prefer having a default media="screen" if nothing specified and maybe the ability in sfc files to do something like:

<style media="screen"> 
</style>

### Importance

nice to have

**Comment by AleCandido** [comment]
I guess in your **Describe the problem** you had some tags that have been compiled...

Try to use backticks to mark them as code: `<style></style>` (generated from <code>\`<style></style>\`</code>).

**Comment by AleCandido** [comment]
However, the `<style></style>` tags inside a page component (a `.svelte` file inside `src/routes`, by default) it's already compiled to something like:

```html
<link rel="stylesheet" href="/_app/assets/pages/<path/to/your/page>.svelte-<hash>.css">
```

and the corresponding stylesheet.

So, I guess, your proposal boils down to just preserve the `media` attribute, moving that from the compiled `<style></style>` to the generated `<link>`.

**Comment by espositofulvio** [comment]
Yes you've got the point but mine is just a suggestion because I just have
a vague idea of how css is generated and I'm not sure if that's really
feasible. Maybe that css can be merged in a single file with other styles
and conflicting media attributes. I don't know, but I guess it would be
something in the direction of accessibility anyway.

**Comment by AleCandido** [comment]
I'm not expert myself, but I just had a look to the compile assets in my `kit` app, and actually the scheme is the following: any page component `<style/>` content is transformed into a `.css` file on its own, with path `_app/assets/pages/<path/to/your/page>.svelte-<hash>.css`, and linked into the corresponding `.html` file.

Since to any `<style/>` there is a corresponding `<link/>`, than I guess it's doable. It's just a matter of will, maybe people prefer to keep `<style/>` attributes to those from the HTML specs, in which you can propose another syntax to achieve the same (even though personally I would prefer to be a tag attribute, and so your suggestion is the most sensible one).

**Comment by Rich-Harris** [comment]
Screen readers rely on the browser's accessibility tree, which takes things like `display: none` and pseudo elements into account. My understanding is that `media="screen"` stylesheets are still generally used (they are _screen_ readers, after all), so this would have no effect; if stylesheets _were_ excluded then the accessibility tree would likely be incorrect.

In short, the `media` attribute should only be used if it was specified by the CSS author. What _would_ be cool is if stylesheets were split up based on media query, i.e. this...

```css
.foo { color: red; }

@media (screen) {
  .bar { color: green; }
}

@media (print) {
  .baz { color: blue; }
}
```

...resulted in this...

```html
<link rel="stylesheet" href="/_app/assets/abc123.css">
<link rel="stylesheet" href="/_app/assets/def456.css" media="screen">
<link rel="stylesheet" href="/_app/assets/ghi789.css" media="print">
```

...but I expect that would require cooperation from Vite.

**Comment by espositofulvio** [comment]
I can see your idea of splitting based on the media query is actually better, surely I second that!
